### PR TITLE
DEV: make regolith work on could based mongodb

### PR DESCRIPTION
**Pull request by st3107**
Hi @sbillinge . I make the regolith work for the cloud based mognodb (also local host mognodb). I test it by building the beamplan from the cloud based rg-db-group in my cloud. Ready for review.

I deprecate the one functionality of the mongoclient in regolith: start the mongo server and close it after the run is finished. The motivation is that (1) starting and closing server take time (2) usually the server will always run in the backend and we don't really need to open and close it every time we run a regolith command (3) if we encounter some error in the run, the server will not be closed and user usually doesn't notice that and when the user tries to run again, it will produce error because regolith will try to start a server that is already running.

### CODE REVIEW

File: regolith/mongoclient.py [hunk 1]
```diff
@@ -102,6 +105,27 @@ def import_yamls(dbpath: str, dbname: str, host: str = None, uri: str = None) ->
     return
 
 
+def load_mongo_col(col: Collection) -> dict:
```
Comment (sbillinge) [inline regolith/mongoclient.py:108]: is there a way to write a test for this?

Comment (st3107) [reply regolith/mongoclient.py:108]: Yes. I will work on it.

**Comment by st3107** [comment]
I tested the mongodb backend on all the builders. It works for all builders except for building figures. I am not sure why. This bug probably takes time to fix. I don't except the mongo backend works for helpers so I didn't add tests for that. This is only the inital step and I think we can first merge it and then start a another step forward.

**Comment by st3107** [comment]
I will skip the figure test for mongo. Not ready for merge.

**Comment by st3107** [comment]
Done.

**Comment by sbillinge** [comment]
thanks @st3107

**Pull request merged**